From 5e84aec87108e0481af7495a1e9a9953d8590d70 Mon Sep 17 00:00:00 2001
From: Xue Liu <liuxuenetmail@gmail.com>
Date: Mon, 6 Feb 2017 17:43:19 +0000
Subject: net: micrel: Disable PME

Disable PME for Micrel phy driver allowing the Ethernet ports LED
driver to work on marduk platform.

Signed-off-by: Xue Liu <liuxuenetmail@gmail.com>
---
 drivers/net/phy/micrel.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/phy/micrel.c b/drivers/net/phy/micrel.c
index ea92d52..73401fb 100644
--- a/drivers/net/phy/micrel.c
+++ b/drivers/net/phy/micrel.c
@@ -273,6 +273,7 @@ static int kszphy_config_init(struct phy_device *phydev)
 	struct kszphy_priv *priv = phydev->priv;
 	const struct kszphy_type *type;
 	int ret;
+	int temp = 0;
 
 	if (!priv)
 		return 0;
@@ -308,6 +309,11 @@ static int kszphy_config_init(struct phy_device *phydev)
 			return ret;
 	}
 
+	/* disable PME */
+	temp = phy_read(phydev, 0x16);
+	temp &= ~(1 << 15);
+	phy_write(phydev, 0x16, temp);
+
 	return 0;
 }
 
-- 
2.7.4

